### PR TITLE
Improvements to Wall Game code

### PR DESCRIPTION
**Pull request by Hazado**
Removed forced camera call in maintainer nexus (Hopefully fixes camera weirdness sometimes experienced)
Blocker panel should be much more responsive with lots of players in the same age
Added clothing removal check when you use quit button or win region
Panel buttons should only respond to player sitting down now
Player sitting down now sets sdl for Panel blockers
Solo Mode added - panel on one side will duplicate itself to other side - will only work when your the only player in the age
Blocker count can be changed at any time before both sides have confirmed the count
Panel lights and blocker wall wont flash anymore when blocker count is changed after confirmation

### CODE REVIEW

File: Scripts/Python/grsnWallPython.py [hunk 1]
```diff
@@ -404,8 +407,9 @@ def OnNotify(self,state,id,events):
                     SouthPanelSound.run(self.key, state='denied')
                 else:
                     self.ChangeGameState(kSouth, kWait)
-                    SouthTubeOpen.run(self.key)
                     SouthPanelSound.run(self.key, state='gameStart')
+                    if not bool(PtGetPlayerList()):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:411]:
```suggestion
                    if not PtGetPlayerList():
```

File: Scripts/Python/grsnWallPython.py [hunk 2]
```diff
@@ -669,10 +677,10 @@ def OnNotify(self,state,id,events):
                     except:
                         PtDebugPrint("grsnWallPython::OnNotify: Blocker not found on either panel")
                         return
-                if(team == kNorth and ageSDL["nState"][0] == kSetBlocker):
-                    self.SetPanelBlocker(kNorth, index, not self.FindBlocker(kNorth, index))
-                if(team == kSouth and ageSDL["sState"][0] == kSetBlocker):
-                    self.SetPanelBlocker(kSouth, index, not self.FindBlocker(kSouth, index))
+                if((team == kNorth or not bool(PtGetPlayerList())) and ageSDL["nState"][0] == kSetBlocker):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:680]:
```suggestion
                if ((team == kNorth or not PtGetPlayerList()) and ageSDL["nState"][0] == kSetBlocker):
```

File: Scripts/Python/grsnWallPython.py [hunk 3]
```diff
@@ -669,10 +677,10 @@ def OnNotify(self,state,id,events):
...
+                    self.SetPanelBlocker(kNorth, index, not self.FindBlocker(kNorth, index), eventAvatar=PtFindAvatar(events))
+                if((team == kSouth or not bool(PtGetPlayerList())) and ageSDL["sState"][0] == kSetBlocker):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:682]:
```suggestion
                if ((team == kSouth or not PtGetPlayerList()) and ageSDL["sState"][0] == kSetBlocker):
```

File: Scripts/Python/grsnWallPython.py [hunk 4]
```diff
@@ -832,72 +840,93 @@ def SetPanelMode(self,team,disableAll=False,onlNorthLights=False):
...
             tenBtn[team].disable()
             fifteenBtn[team].disable()
-        elif(state == kSit):
-            goBtnObject[team].value.runAttachedResponder(kBright)
-            goButton[team].enable()
+        elif(state == kStandby or state == kEnd):
+            for i in range(0,20):
+                CountLights[team].value[i].runAttachedResponder(kRedOff)
+            for i in range(0,171):
+                Panel[team].value[i].physics.disable()
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:869]: Are these loops iterating all of the lights and panel objects?

File: Scripts/Python/grsnWallPython.py [hunk 5]
```diff
@@ -832,72 +840,93 @@ def SetPanelMode(self,team,disableAll=False,onlNorthLights=False):
...
+            #goBtnObject[team].value.runAttachedResponder(kDim)
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:870]:
```suggestion
```

File: Scripts/Python/grsnWallPython.py [hunk 6]
```diff
@@ -832,72 +840,93 @@ def SetPanelMode(self,team,disableAll=False,onlNorthLights=False):
...
-            goBtnObject[team].value.runAttachedResponder(kDim)
-            goButton[team].disable()
-            upButton[team].enable()
-            dnButton[team].enable()
-            readyButton[team].enable()
-            fiveBtn[team].enable()
-            tenBtn[team].enable()
-            fifteenBtn[team].enable()
+        elif(state == kSit):
+            #goBtnObject[team].value.runAttachedResponder(kBright)
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:879]:
```suggestion
```

File: Scripts/Python/grsnWallPython.py [hunk 7]
```diff
@@ -832,72 +840,93 @@ def SetPanelMode(self,team,disableAll=False,onlNorthLights=False):
...
+                readyButton[team].disable()
+                fiveBtn[team].disable()
+                tenBtn[team].disable()
+                fifteenBtn[team].disable()
+        elif(ageSDL["nState"][0] == kSelectCount or ageSDL["sState"][0] == kSelectCount):
+            if ageSDL["NumBlockers"][0] == 0:
+                for i in range(0,171):
+                    self.SetPanelBlocker(team, i, False, onlyLight=True)
+                    Panel[team].value[i].physics.disable()
+            #goBtnObject[team].value.runAttachedResponder(kDim)
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:893]:
```suggestion
```

File: Scripts/Python/grsnWallPython.py [hunk 8]
```diff
@@ -768,11 +779,12 @@ def OnSDLNotify(self,VARname,SDLname,playerID,tag):
                 #South player is ready - transmit blockers to the wall
                 for blocker in ageSDL["southWall"]:
                     SouthWall.value[blocker].physics.enable()
-                if(ageSDL["nState"][0] >= kWait and eventHandler):
+                if(ageSDL["nState"][0] >= kWait and eventHandler and bool(PtGetPlayerList())):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:782]:
```suggestion
                if (ageSDL["nState"][0] >= kWait and eventHandler and PtGetPlayerList()):
```

File: Scripts/Python/grsnWallPython.py [hunk 9]
```diff
@@ -768,11 +779,12 @@ def OnSDLNotify(self,VARname,SDLname,playerID,tag):
...
                     eventHandler.Handle(kEventEntry)
-                    
+                SouthTubeOpen.run(self.key)
+
         if(VARname == "nWallPlayer" and ageSDL["nWallPlayer"] != (-1,)):
-            if(ageSDL["sState"][0] == kEntry):
+            if(ageSDL["sState"][0] == kEntry or not bool(PtGetPlayerList())):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:787]:
```suggestion
            if (ageSDL["sState"][0] == kEntry or not PtGetPlayerList()):
```

File: Scripts/Python/grsnWallPython.py [hunk 10]
```diff
@@ -785,7 +797,7 @@ def OnSDLNotify(self,VARname,SDLname,playerID,tag):
                 PtGetLocalAvatar().avatar.runBehaviorSetNotify(NorthTubeMulti.value, self.key, NorthTubeMulti.netForce)
 
         if(VARname == "sWallPlayer" and ageSDL["sWallPlayer"] != (-1,)):
-            if(ageSDL["nState"][0] == kEntry):
+            if(ageSDL["nState"][0] == kEntry or not bool(PtGetPlayerList())):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:800]:
```suggestion
            if (ageSDL["nState"][0] == kEntry or not PtGetPlayerList()):
```

File: Scripts/Python/grsnWallPython.py [hunk 11]
```diff
@@ -832,72 +840,93 @@ def SetPanelMode(self,team,disableAll=False,onlNorthLights=False):
...
-            tenBtn[team].disable()
-            fifteenBtn[team].disable()
             for i in range(0,ageSDL["NumBlockers"][0]):
                 if(i < self.GetNumBlockerSet(team)):
                     CountLights[team].value[i].runAttachedResponder(kTeamLightsOn)
                 else:
                     CountLights[team].value[i].runAttachedResponder(kRedOn)
             for i in range(0,171):
                 Panel[team].value[i].physics.enable()
+            #goBtnObject[team].value.runAttachedResponder(kPulse)
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:915]:
```suggestion
```

File: Scripts/Python/grsnWallPython.py [hunk 12]
```diff
@@ -832,72 +840,93 @@ def SetPanelMode(self,team,disableAll=False,onlNorthLights=False):
...
+                readyButton[team].disable()
+                fiveBtn[team].disable()
+                tenBtn[team].disable()
+                fifteenBtn[team].disable()
         else:
-            goBtnObject[team].value.runAttachedResponder(kDim)
-            goButton[team].disable()
             for i in range(0,171):
                 Panel[team].value[i].physics.disable()
+            #goBtnObject[team].value.runAttachedResponder(kDim)
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:927]:
```suggestion
```

File: Scripts/Python/grsnWallPython.py [hunk 13]
```diff
@@ -995,15 +1024,19 @@ def RequestGameUpdate(self):
...
             if(on):
                 Lights[team].value[id].runAttachedResponder(kTeamLightsOn)
             else:
                 Lights[team].value[id].runAttachedResponder(kTeamLightsOff)
             return
 
+        ageSDL = PtGetAgeSDL()
+
+        if not (PtGetClientIDFromAvatarKey(eventAvatar.getKey()) == ageSDL["nChairOccupant"][0] or PtGetClientIDFromAvatarKey(eventAvatar.getKey()) == ageSDL["sChairOccupant"][0]):
+            return
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:1038]:
```suggestion
        cID = PtGetClientIDFromAvatarKey(eventAvatar.getKey()
        if cID in {ageSDL["nChairOccupant"][0], ageSDL["sChairOccupant"][0]}:
            return
```

File: Scripts/Python/grsnWallPython.py [hunk 14]
```diff
@@ -472,10 +476,14 @@ def OnNotify(self,state,id,events):
         if(id == readyButtonNorth.id and not state and ageSDL["nState"][0] == kSelectCount):
             self.ChangeGameState(kNorth, kSetBlocker)
             NorthPanelSound.run(self.key, state='up')
+            if not bool(PtGetPlayerList()):
+                self.ChangeGameState(kSouth, kSetBlocker)
             return
         if(id == readyButtonSouth.id and not state and ageSDL["sState"][0] == kSelectCount):
             self.ChangeGameState(kSouth, kSetBlocker)
             SouthPanelSound.run(self.key, state='up')
+            if not bool(PtGetPlayerList()):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:485]:
```suggestion
            if not PtGetPlayerList():
```

File: Scripts/Python/grsnWallPython.py [hunk 15]
```diff
@@ -472,10 +476,14 @@ def OnNotify(self,state,id,events):
         if(id == readyButtonNorth.id and not state and ageSDL["nState"][0] == kSelectCount):
             self.ChangeGameState(kNorth, kSetBlocker)
             NorthPanelSound.run(self.key, state='up')
+            if not bool(PtGetPlayerList()):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:479]:
```suggestion
            if not PtGetPlayerList():
```

File: Scripts/Python/grsnWallPython.py [hunk 16]
```diff
@@ -387,8 +389,9 @@ def OnNotify(self,state,id,events):
                     NorthPanelSound.run(self.key, state='denied')
                 else:
                     self.ChangeGameState(kNorth, kWait)
-                    NorthTubeOpen.run(self.key)
                     NorthPanelSound.run(self.key, state='gameStart')
+                    if not bool(PtGetPlayerList()):
```
Comment (Hoikas) [inline Scripts/Python/grsnWallPython.py:393]:
```suggestion
                    if not PtGetPlayerList():
```

**Review by Hoikas — COMMENTED** [review]
First pass at reviewing this. It would be really nice if the changes were grouped into atomic commits by topic such that I could better understand what exactly each set of changes is designed to fix.

File: Scripts/Python/grsnWallPython.py (hunk 1 above)
Comment (Hazado) [reply Scripts/Python/grsnWallPython.py:411]: This check is for the Solo Mode wall game.
PtGetPlayerList() returns a list of players, without the bool conversion it wont properly work.

File: Scripts/Python/grsnWallPython.py (hunk 4 above)
Comment (Hazado) [reply Scripts/Python/grsnWallPython.py:869]: Yes, they turn on/off the lights or physics for the panels

File: Scripts/Python/grsnWallPython.py (hunk 1 above)
Comment (ZarothYe) [reply Scripts/Python/grsnWallPython.py:411]: Empty list is False-y and populated list is True-thy, so Hoikas is correct that the cast is not needed. Tested in python REPL:
```
>>> foo = []
>>> if not foo:
...     print("x")
...
x
>>> foo = [1]
>>> if not foo:
...     print("x")
...
>>>
```

**Review by Hoikas — APPROVED** [review]

**Pull request merged**